### DISCUSSION TITLE
Tuya PC321-TY Power Clamp

### DISCUSSION THREAD

**Discussion by z-master42** [jasonacox/tinytuya#483] [Q&A] [answered]
I have this power clamp, which works so far. Unfortunately, it happens that it is suddenly no longer accessible.
Everything is available and accessible in the Tuya app.
I can also see all the data for the device via tinytuya, but there is no response when polling.
The IP can be pinged.
Any ideas what the problem could be?

![grafik](https://github.com/jasonacox/tinytuya/assets/66380371/987f4739-230a-45d5-b1d9-eddaa339b7f7)

**Comment by jasonacox** [comment]
Hi @z-master42 

> I can also see all the data for the device via tinytuya, but there is no response when polling.

It could be a bad local key. Turn on debug mode to see if you get any more clues:

```
python -m tinytuya scan -d
```

**Reply by z-master42** [reply]
Hey,

```
DEBUG:Listening for Tuya devices on UDP ports 6666, 6667 and 7000
DEBUG:Received valid UDP packet: {'ip': '192.168.42.33', 'gwId': 'bfdc9374148c1c4dcfq29r', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': '9rjjf6rdkca9woju', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa000000010000000300000044621e967fc591884c612670a193ae1668847ca439847272b3ea714406ff212c0604d187f97685ebb2a4667664d5b908af3a5185e935b88c35c53f526946e7bffb0000aa55'
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa000000010000000300000044621e967fc591884c612670a193ae1668847ca439847272b3ea714406ff212c0604d187f97685ebb2a4667664d5b908af3a5185e935b88c35c53f526946e7bffb0000aa55'
DEBUG:Received valid UDP packet: {'ip': '192.168.42.33', 'gwId': 'bfdc9374148c1c4dcfq29r', 'active': 2, 'ablilty': 0, 'encrypt': True, 'productKey': '9rjjf6rdkca9woju', 'version': '3.4', 'token': True, 'wf_cfg': True}
DEBUG:final payload: b'0123456789abcdef'
DEBUG:payload encrypted=b'000055aa000000010000000300000044621e967fc591884c612670a193ae1668847ca439847272b3ea714406ff212c0604d187f97685ebb2a4667664d5b908af3a5185e935b88c35c53f526946e7bffb0000aa55'
```

I can't see anything "wrong" so far.
The strange thing is that the device was previously accessible with the local key and now suddenly not anymore.

**Reply by jasonacox** [reply]
> The strange thing is that the device was previously accessible with the local key and now suddenly not anymore.

Ah! That's good to know. Did you happen to re-pair this Tuya device with SmartLife app? If so, the local key has been reset.  

Just to be sure, you can run `python -m tinytuya wizard` again to generate a new devices.json.

**Reply by z-master42** [reply]
No, not knowingly. Everything is unchanged in the app.

> Just to be sure, you can run python -m tinytuya wizard again to generate a new devices.json.

I have already done that. All data is unchanged and there is also no response when polling.

**Comment by uzlonewolf** [comment, marked as answer]
Have you tried turning the device completely off and back on?  I don't know about that device specifically, but I have others that occasionally "forget" what their key is supposed to be and need to be power-cycled before they'll start working again.  I suspect it's memory corruption somewhere but haven't been able to dump the memory from one while it is happening.

**Reply by z-master42** [reply]
Haha,

that was too obvious.
I hadn't actually considered that, as the device was still accessible as normal in the Tuya app.

But now it's working properly again.

Thank you very much
